hw1/02-palindrome.js

```diff
@@ -1,3 +1,36 @@
-const elem = document.querySelector('input');
+document.addEventListener('DOMContentLoaded', () => {
+    console.log('DOMContentLoaded event fired'); // Debugging
 
-elem.addEventListener('input', handleInput);
+    const numberInput = document.getElementById('numberInput');
+    const checkPalindromeBtn = document.getElementById('checkPalindromeBtn');
+    const resultParagraph = document.getElementById('result');
+
+    checkPalindromeBtn.addEventListener('click', () => {
+
+        const inputValue = parseInt(numberInput.value, 10);
+
+        if (isNaN(inputValue) || inputValue < 0) {
+            resultParagraph.textContent = 'Please enter a positive number.';
+            resultParagraph.classList.add('text-danger');
+            return;
+        }
+
+        if (isPalindrome(inputValue.toString())) {
+            resultParagraph.textContent = `Yes. This is a palindrome!`;
+            resultParagraph.classList.remove('text-danger');
+            resultParagraph.classList.add('text-success');
+        } else {
+            resultParagraph.textContent = `No. Try again`;
+            resultParagraph.classList.remove('text-success');
+            resultParagraph.classList.add('text-danger');
+        }
+    });
+
+    function isPalindrome(number) {
+        let reversed = number.split('').reverse().join('');
```

Comment:
Nice solution! Considered using a string solution, but wanted to challenge my algorithm skills 😆